### PR TITLE
HDDS-9236. Fix snapdiff output for key modification

### PR DESCRIPTION
**Pull request by swamirishi**
## What changes were proposed in this pull request?

Currently snapdiff checks only key versions to check if a key has been modified. We have to check if other fields have also changed like (Acl changes, replication config changes etc.)

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9236

## How was the patch tested?
Integration tests

### CODE REVIEW

**Comment by swamirishi** [comment]
@hemantk-12 @smengcl Please take a look at the changes

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java [hunk 1]
```diff
@@ -760,6 +760,41 @@ public String getObjectInfo() {
         '}';
   }
 
+
+  public boolean isKeyInfoSame(OmKeyInfo omKeyInfo, boolean checkPath,
```
Comment (hemantk-12) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java:764]: I think `isKeyInfoSame` is very specific to snapshot diff and should be added to snapshot util.

**Review by hemantk-12 — COMMENTED** [review]
Thanks for the patch @swamirishi 

Overall looks good to me.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java [hunk 2]
```diff
@@ -1468,7 +1468,9 @@ private boolean isObjectModified(String fromObjectName, String toObjectName,
     final WithObjectID fromObject = fromSnapshotTable.get(fromObjectName);
     final WithObjectID toObject = toSnapshotTable.get(toObjectName);
     if ((fromObject instanceof OmKeyInfo) && (toObject instanceof OmKeyInfo)) {
-      return !SnapshotDeletingService.isBlockLocationInfoSame(
+      return !((OmKeyInfo) fromObject).isKeyInfoSame((OmKeyInfo) toObject,
+          false, false, false, false)
+          || !SnapshotDeletingService.isBlockLocationInfoSame(
           (OmKeyInfo) fromObject, (OmKeyInfo) toObject);
```
Comment (smengcl) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java:1474]: Ideally this should be one check for the key.

Would you wrap this in a reasonably-named helper method for readability?

Comment (swamirishi) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java]: done

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java (hunk 1 above)
Comment (swamirishi) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java:764]: It is better this way to put in key info if some field is added in omKeyInfo it should impact the equals method as well thus impacting snapshot diff.

Comment (hemantk-12) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java:764]: Fair enough.

**Comment by hemantk-12** [comment]
Thank @swamirishi for working on this and @smengcl for reviewing.

**Pull request merged**